tests/components/mqtt/test_discovery.py

```diff
@@ -1487,6 +1487,7 @@ async def test_mqtt_integration_discovery_subscribe_unsubscribe(
         await async_start(hass, "homeassistant", entry)
         await hass.async_block_till_done()
         await hass.async_block_till_done()
+        await hass.async_block_till_done()
```

Comment:
There is one less iteration of the event loop happening now so we need to block one more time.